### PR TITLE
DCON-4806 harden merge request copying and attachment file-id handling (4/4)

### PR DESCRIPTION
**Pull request by alvinhenrick**
## Jira
[DCON-4806](https://icanbwell.atlassian.net/browse/DCON-4806) (4 of 4 grouped PRs — see ticket for the full finding list; this PR covers merge.js, create.js, and update.js only)

## Summary
Data-mutation and attachment-handling hardening in `src/operations/merge/merge.js`, `src/operations/create/create.js`, and `src/operations/update/update.js`, plus a new `removeFileIdFieldRecursive` helper in `src/utils/removeUnderscoreFields.js`. See the linked Jira ticket for the full technical writeup.

## Tests
Added coverage in `removeUnderscoreFields.test.js` (new helper), `create.test.js`, `update.test.js` (both verify the helper is wired in before attachment handling), and `merge.test.js` (verifies a deep copy is passed to the validator, not the original request body reference).

## Verification
- `eslint` clean.
- `jest --config jest.unit.config.js`: confirmed via `git stash` comparison — same 2 pre-existing failures in `merge.nullSafety.test.js` (unrelated: null-headers and empty-mergeResults edge cases already broken on main), 8 new tests added and passing. Full suite: 329 failing before and after, unchanged. Zero regressions.
- Full Docker-backed `jest.config.js` suite not run locally (no Docker in the authoring environment) — please confirm CI is green before merging.

[DCON-4806]: https://icanbwell.atlassian.net/browse/DCON-4806?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/tests/unit/operations/create/create.test.js [hunk 1]
```diff
@@ -259,6 +259,57 @@ describe('CreateOperation', () => {
             expect(mocks.databaseBulkInserter.executeAsync).toHaveBeenCalled();
         });
 
+        // DCON-4806: a client-supplied _file_id (or any other internal underscore field,
+        // e.g. _uuid) must be stripped from the raw payload before the Resource is even
```
Comment (alvinhenrick) [inline src/tests/unit/operations/create/create.test.js:263]: Nice.

**Review by alvinhenrick — COMMENTED** [review]
Thank ou for adding test - Very helpful

**Review by mintusah25 — APPROVED** [review]
